homeassistant/components/aseko_pool_live/sensor.py

```diff
@@ -2,77 +2,107 @@
+        value_fn=lambda unit: unit.cl_free,
+    ),
+    AsekoSensorEntityDescription(
+        key="ph",
+        device_class=SensorDeviceClass.PH,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda unit: unit.ph,
+    ),
+    AsekoSensorEntityDescription(
+        key="redox",
+        translation_key="redox",
+        native_unit_of_measurement=UnitOfElectricPotential.MILLIVOLT,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda unit: unit.redox,
+    ),
+    AsekoSensorEntityDescription(
+        key="salinity",
+        translation_key="salinity",
+        native_unit_of_measurement="kg/m³",
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda unit: unit.salinity,
+    ),
+    AsekoSensorEntityDescription(
+        key="water_temperature",
+        translation_key="water_temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda unit: unit.water_temperature,
+    ),
+]
+
+
 async def async_setup_entry(
     hass: HomeAssistant,
     config_entry: ConfigEntry,
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up the Aseko Pool Live sensors."""
-    data: list[tuple[Unit, AsekoDataUpdateCoordinator]] = hass.data[DOMAIN][
+    data: tuple[str, AsekoDataUpdateCoordinator] = hass.data[DOMAIN][
         config_entry.entry_id
     ]
+    user_id, coordinator = data
+    units = coordinator.data.values()
+    for unit in units:
+        async_add_entities(
+            AsekoSensorEntity(unit, user_id, coordinator, description)
+            for description in SENSORS
+            if getattr(unit, description.key) is not None
+        )
+
+
+class AsekoSensorEntity(AsekoEntity, SensorEntity):
+    """Representation of an Aseko unit sensor entity."""
 
-    async_add_entities(
-        VariableSensorEntity(unit, variable, coordinator)
-        for unit, coordinator in data
-        for variable in unit.variables
-    )
-
-
-class VariableSensorEntity(AsekoEntity, SensorEntity):
-    """Representation of a unit variable sensor entity."""
-
-    _attr_state_class = SensorStateClass.MEASUREMENT
-
-    def __init__(
-        self, unit: Unit, variable: Variable, coordinator: AsekoDataUpdateCoordinator
-    ) -> None:
-        """Initialize the variable sensor."""
-        super().__init__(unit, coordinator)
-        self._variable = variable
-
-        translation_key = {
-            "Air temp.": "air_temperature",
-            "Cl free": "free_chlorine",
-            "Water temp.": "water_temperature",
-        }.get(self._variable.name)
-        if translation_key is not None:
-            self._attr_translation_key = translation_key
-        else:
-            self._attr_name = self._variable.name
-
-        self._attr_unique_id = f"{self._unit.serial_number}{self._variable.type}"
```

Comment:
the old variable type used to be `airTemp` and `waterTemp` among other things. We now changed that to be `air_temperature` and `water_temperature`. Even though the api changed, we should try to maintain the same unique_id for the same datapoints.

Reply:
Fixed in https://github.com/home-assistant/core/pull/126133/commits/12b547e63814a2b2b3f15de291e6305d3938043a

Reply:
Fixed for `airTemp`, `free_chlorine`, `ph`, `rx`, `waterTemp` and `water_flow`. I don't know what the `variable.type` of `salinity` was, so I can't be sure this wasn't changed.